aptos-move/framework/aptos-framework/sources/aptos_governance.move

```diff
@@ -375,6 +375,15 @@ module aptos_framework::aptos_governance {
         reconfiguration::reconfigure();
     }
 
+    /// Return the voting power a stake pool has with respect to governance proposals.
+    fun get_voting_power(pool_address: address): u64 {
+        let (active, _, pending_active, pending_inactive) = stake::get_stake(pool_address);
+        // We calculate the voting power as total non-inactive stakes of the pool. Even if the validator is not in the
+        // active validator set, as long as they have a lockup (separately checked in create_proposal and voting), their
+        // stake would still count in their voting power for governance proposals.
+        active + pending_active + pending_inactive
```

Comment:
Is there a world where I can stake, vote, unstake, within the same epoch? Ie can I go from pending active to inactive via user txn

Reply:
No that's not possible because of the lockup check. In order to vote, you need to have a remaining lockup longer than the duration of the vote. So as long as the voting duration is longer than an epoch, that's not possible.